tests/test_main.py

```diff
@@ -103,3 +104,17 @@ def mocked_main(*args):
 
     with pytest.raises(FakeFailureError):
         xonsh.main.main()
+
+
+def test_xonsh_failback_script_from_file(shell, monkeypatch):
```

Comment:
there is also the case where we want to run a script and then go into interactive
mode like the `-i` option of python, something that doesn't work atm and
 it didn't work the last time i checked way back before the failback PR.

maybe if we isolate the try/except that calls the failback function 
to the `XonshMode.interactive` clause of main, we can then just mock out
`execlp` and call directly the `_failback_to_other_shells` function in tests.